molgenis-js/src/main/java/org/molgenis/js/magma/JsMagmaScriptEvaluator.java

```diff
@@ -97,16 +120,25 @@ private static Object toScriptEngineValue(Entity entity, Attribute attr)
 			case FILE:
 			case XREF:
 				Entity xrefEntity = entity.getEntity(attrName);
-				value = xrefEntity != null ? toScriptEngineValue(xrefEntity,
-						xrefEntity.getEntityType().getIdAttribute()) : null;
+				value = toScriptEngineValueMap(xrefEntity, depth - 1);
 				break;
 			case CATEGORICAL_MREF:
 			case MREF:
 			case ONE_TO_MANY:
-				Iterable<Entity> mrefEntities = entity.getEntities(attrName);
-				value = stream(mrefEntities.spliterator(), false).map(
-						mrefEntity -> toScriptEngineValue(mrefEntity, mrefEntity.getEntityType().getIdAttribute()))
-																 .collect(toList());
+				ScriptObjectMirror jsArray = null;
+				try
+				{
+					jsArray = (ScriptObjectMirror) jsScriptEngine.eval("var arr = []; arr");
+					@SuppressWarnings("unchecked")
+					List<Object> mrefValues = jsArray.to(List.class);
+					entity.getEntities(attrName)
+						  .forEach(mrefEntity -> mrefValues.add(toScriptEngineValueMap(mrefEntity, depth - 1)));
+				}
+				catch (javax.script.ScriptException ex)
+				{
+					// ignore
```

Comment:
is this a good idea ?, why not handle it , a the least add doc why this is ignored